### DISCUSSION TITLE
How to add logo through web settings

### DISCUSSION THREAD

**Discussion by Basserra** [kimai/kimai#2416] [Questions & Answers]
I've been having some troubles adding a logo to my Kimai setup. I have installed Kimai directly on my Debian host, not through docker. I've added my logo to .../assets/images/Logo.png and it's owned by the server user like the rest of the files. In the web settings page I've tried the full URL and IP address and localhost, and with/without the custom port (e.g. "https://kimai.mstarm.net:20443/assets/images/Logo.png"). The subdomain is not publicly available, its internal only, and it's in the hosts file pointing to 127.0.0.1 as the local DNS server points to a reverse proxy.

How might I go about adding a logo through the web settings page correctly? Or does it have to be configured with local.yaml? Could I also use a system path?

**Comment by kevinpapst** [comment]
https://www.kimai.org/documentation/branding.html
You can do all in the System configuration, no need for local.yaml
Post an image of your current config if you need further help.

**Reply by Basserra** [reply]
Not sure which part of config you're talking about, so here's what I'm seeing and nginx.conf which is a mess right now, sorry. I also haven't configured a local.yaml yet, as I haven't needed it.
All has been well and functional thus far with the exception of the logo, email and api setup for the iOS app; which I haven't gotten into yet. I migrated the database from a docker setup that was essentially the same.

![image](https://user-images.githubusercontent.com/53457028/110676088-d1333580-8190-11eb-87e0-62401842881a.png)
![image](https://user-images.githubusercontent.com/53457028/110676206-f32cb800-8190-11eb-8ebc-d61690aec158.png)
![image](https://user-images.githubusercontent.com/53457028/110676360-19525800-8191-11eb-9a54-b3f3cb487370.png)
![image](https://user-images.githubusercontent.com/53457028/110677404-5e2abe80-8192-11eb-8d04-04840925984a.png)

Thanks for your time

**Reply by kevinpapst** [reply]
You have to move the image, that's all. See your nginx `root` points to `public/`, thats what essentially maps to the domain `kimai.mstarm.net`. 

The rest is up to you, eg. move the image to `public/Logo.png` and change the Logo config to `/Logo.png`.

**Reply by Basserra** [reply]
You're right! Moved it under the public folder, changed the path, and it displayed correctly immediately. Thank you much for your help!

Next I'd like to get the app working, but I'll attempt to figure it out on my own before I start a new discussion thread; unless you might have anything to offer to me about it now. The app can pull data from the server, but it can't push the entries. I get this line from /api/timesheets `"Authentication required, missing headers: X-AUTH-USER, X-AUTH-TOKEN"`

**Reply by kevinpapst** [reply]
Sounds weird, sorry - no idea.
Test the API by itself, maybe you have a problem with PATCH requests in your setup?
